mimic/test/test_nova.py

```diff
@@ -35,20 +34,67 @@ def status_of_server(test_case, server_id):
     return get_server_response_body['server']['status']
 
 
-def quick_create_server(helper, region="ORD"):
+def server_args(name=None, imageRef=None, flavorRef=None,
+                metadata=None, diskConfig=None):
+    """
+    Helper function to generate a create server request body.
+
+    :param name: Name of the server
+    :param imageRef: Image of the server
+    :param flavorRef: Flavor size of the server
+    :param metadata: Metadata of the server
+    :param diskConfig: the "OS-DCF:diskConfig" setting for the server
+
+    :return: A stringified JSON dictionary that can be passed to a create
+        server request.
+    """
+    data = {
+        "name": name or 'test_server',
+        "imageRef": imageRef or "test-image",
+        "flavorRef": flavorRef or "test-flavor"
+    }
+    if metadata is not None:
+        data['metadata'] = metadata
+    if diskConfig is not None:
+        data["OS-DCF:diskConfig"] = diskConfig
+    return json.dumps({"server": data})
+
+
+def create_server(helper, body, region="ORD", json=False):
+    """
+    Create a server with the given body and returns the response object.
+
+    :param str body: String containing the server args
+    :param str region: The region in which to create the server
+    :param bool json: Whether to use json_request
+
+    :return: either the response object, or the response object and JSON
+        body if ``json`` is `True`.
+    """
+    request_func = json_request if json else request
+    create_server = request_func(
+        helper.test_case,
+        helper.root,
+        "POST",
+        '{0}/servers'.format(helper.get_service_endpoint(
+            "cloudServersOpenStack", region)),
+        body
+    )
+    return helper.test_case.successResultOf(create_server)
+
+
+def quick_create_server(helper, region="ORD", name=None):
```

Comment:
Now that there's another kind of server creation to contrast it with, "quick"ness doesn't seem to be its defining property. I am thinking that this should be something like `id_for_new_server`?